lib/internal/dns/promises.js

```diff
@@ -112,7 +112,9 @@ function lookup(hostname, options) {
     hints = options.hints >>> 0;
     family = options.family >>> 0;
     all = options.all === true;
-    verbatim = options.verbatim === true;
+    if (typeof options.verbatim === 'boolean') {
+      verbatim = options.verbatim === true;
+    }
```

Comment:
Would it be more useful to throw if the value is not a boolean rather than ignoring it? It the value is not a boolean, it's probably a mistake that users would want to spot.
```suggestion
    if (options.verbatim !== undefined) {
      validateBoolean(options.verbatim, 'options.verbatim');
      verbatim = options.verbatim;
    }
```

Reply:
Yes, that's better. But I'm afraid of introducing a breaking change here. Let's reconsider this in the future.

Reply:
I think it's OK to do it in a possible breaking way as long as we don't land it in v15.x. Then we can backport it to v14.x (and v12.x?) without the breaking part.

Reply:
@mcollina wdyt? Should this land with or without argument validation (possibly breaking change)?

Reply:
I'm going to say I prefer the argument validation.

Reply:
I still want to keep the possibility to backport this to v12 and v14.

Reply:
We could still backport it without the argument validation. I volunteer to take care of the backport PR if that if that helps.